### PR TITLE
add isRestore parameter to setState

### PR DESCRIPTION
**Pull request by btamas**
Related to https://oat-sa.atlassian.net/browse/nex-1260

Differentiate inital item state restore and normal setState

Test: `npm run test`

### CODE REVIEW

**Review by krampstudio — CHANGES_REQUESTED** [review]
It looks good, but can you please add a test case in https://github.com/oat-sa/tao-item-runner-fe/blob/master/test/runner/api/test.js#L443 ?

File: src/runner/api/itemRunner.js [hunk 1]
```diff
@@ -360,11 +360,12 @@ var itemRunnerFactory = function itemRunnerFactory(providerName, data, options)
          * This should have the effect to restore the item state.
          *
          * @param {Object} state - the new state
+         * @param {boolean} isInitialStateRestore - state restoring or not
          * @returns {itemRunner}
          *
          * @fires itemRunner#error if the state type doesn't match
          */
-        setState: function(state) {
+        setState: function(state, isInitialStateRestore) {
```
Comment (oatymart) [inline src/runner/api/itemRunner.js]: I would have considered adding a default value, `isInitialStateRestore = false`, but if you're sure you found all the uses of this function, this comment can be safely ignored.

**Comment by btamas** [comment]
> It looks good, but can you please add a test case in https://github.com/oat-sa/tao-item-runner-fe/blob/master/test/runner/api/test.js#L443 ?

I wanted to but I do not know how. How can I access dummy item runner from test? My only idea was to set a variable to window, like `window.wasInitialSetStateCall = true` when it was true. Do you have a better idea?

File: src/runner/api/itemRunner.js (hunk 1 above)
Comment (btamas) [reply src/runner/api/itemRunner.js]: added

**Review by oatymart — APPROVED** [review]

**Review by krampstudio — APPROVED** [review]

**Pull request merged**